OMCompiler/Compiler/BackEnd/Tearing.mo

```diff
@@ -320,23 +330,28 @@ algorithm
       BackendDAE.JacobianType jacType;
       Boolean mixedSystem;
 
-    case ((BackendDAE.EQUATIONSYSTEM(eqns=eindex, vars=vindx, jac=BackendDAE.FULL_JACOBIAN(ojac), jacType=jacType, mixedSystem=mixedSystem)), _, _, _) equation
-      true = BackendDAEUtil.getLinearfromJacType(jacType);
-      maxSize = Flags.getConfigInt(Flags.MAX_SIZE_LINEAR_TEARING);
-      if intGt(listLength(vindx), maxSize) and not
-         // always apply tearing if maxSize > 0 and dense matrices are used
-         (intGt(maxSize, 0) and stringEqual(Config.simCodeTarget(), "Cpp") and
-          stringEqual(Flags.getConfigString(Flags.MATRIX_FORMAT), "dense"))
+    // Tearing of linear systems
+    case ((BackendDAE.EQUATIONSYSTEM(eqns=eindex, vars=vindx, jac=BackendDAE.FULL_JACOBIAN(ojac), jacType=jacType, mixedSystem=mixedSystem)), _, _, _)
+      guard BackendDAEUtil.getLinearfromJacType(jacType) algorithm
+
+      // Check if component size > maxSizeLinearTearing
+      maxSize := Flags.getConfigInt(Flags.MAX_SIZE_LINEAR_TEARING);
+      if listLength(vindx) > maxSize and
+        not (stringEqual(Config.simCodeTarget(), "Cpp") and stringEqual(Flags.getConfigString(Flags.MATRIX_FORMAT), "dense"))
       then
         Error.addMessage(Error.MAX_TEARING_SIZE, {intString(strongComponentIndexOut), intString(listLength(vindx)),"linear",intString(maxSize)});
-        fail();
+        (oComp, outRunMatching) := (inComp, false);
+        break;
```

Comment:
This should work as intended, but i think it would be easier to understand and maintain if we combined all the cases. (i do not really like the `break;`) Example:

This is currently:
```
if <COND1> then
   <ERROR1>
   break;
end if;

if <COND2> then
   <ERROR2>
   break;
end if;

<REST>
```

What i would do:
```
if <COND1> then
  <ERROR1>
elseif <COND2> then
  <ERROR2>
else
  <REST>
end if;
```

Probably nitpicking :)

Reply:
> Probably nitpicking :)

Nahh, thats exactly what I wanted to know. The second way should be more clear, I'll try it.